### DISCUSSION TITLE
Any plans to move `containerd-rootless.sh` etc to containerd/containerd?

### DISCUSSION THREAD

**Discussion by 06kellyjac** [containerd/nerdctl#1475] [Q&A] [answered]
Since these things just set up containerd to run rootless and in theory can be used *without* `nerdctl` are there any plans to move these extras to the main containerd/containerd repo?

**Comment by AkihiroSuda** [comment, marked as answer]
Unlikely, as the migration cost is not negligible and containerd-rootless-setuptool.sh contains a bunch of non-core extensions

**Reply by AkihiroSuda** [reply]
Btw downstream packages may package it as `containerd-rootless-extras` without dependency on `nerdctl`.

**Reply by 06kellyjac** [reply]
I was going to call it `containerd-rootless` or `containerd-rootless-helpers` but I guess we could call it `containerd-rootless-extras` to keep things consistent.

We only actually use [containerd-rootless.sh](https://github.com/containerd/nerdctl/blob/master/extras/rootless/containerd-rootless.sh) because we configure the user service & buildkit via nix config

---

What do you mean by non-core extensions? `rootlesskit` & `slirp4netns`?
All that is documented in the [hardway](https://github.com/containerd/containerd/blob/main/docs/rootless.md#hard-way) as things you need to use anyway?

---

If you're interested in the ongoing work (since I've seen you playing with `nix` occasionally :sweat_smile:) it's here:
https://github.com/NixOS/nixpkgs/pull/198991
https://github.com/06kellyjac/nixpkgs/commits/nixos/containerd

**Reply by AkihiroSuda** [reply]
> What do you mean by non-core extensions?

Mostly I meant BuildKit, IPFS, etc.

> If you're interested in the ongoing work (since I've seen you playing with nix occasionally 😅) it's here:

Thanks, will take a look :+1:

**Reply by 06kellyjac** [reply]
Well since buildkit and ipfs aren't part of `containerd-rootless.sh` could just that part move to containerd? :sweat_smile:
I guess the two scripts complement each-other so maybe not
